src/Api/ApiResourceRegistrar.php

```diff
@@ -75,24 +89,52 @@ public function resource(string $resourceName, string $controller, array $option
                 }
 
                 foreach ($actions as $action => $i) {
-                    $routes[$action] = ['verb' => $verb, 'route' => '/'.$action];
+                    $routes[$action] = ['verb' => $verb, 'route' => '/' . $action];
                 }
             }
         }
 
+        $mapping = [];
+
+        if ($modelClass) {
+            $modelName = lcfirst($modelName ?? $this->resolveModelClass($modelClass));
+            $mapping[$modelName] = $modelClass;
+        }
+
         foreach ($routes as $action => $opt) {
             $verb = $opt['verb'];
-            $this->api->$verb($resourceName.$opt['route'], [
-                'as' => trim($resourceName.'.'.$action),
-                'uses' => $controller.'@'.$action
+            $route = $opt['route'];
+            if ($modelName) {
+                $route = str_replace('{id}', "{{$modelName}}", $opt['route']);
```

Comment:
isn't better to build valid route from beginning instead of replacing them in loop? For example with `getVerbsRoutesConfig(string $attribute = '{id}'): array` ?